source/helpers/render-conventional-commit-types.ts

```diff
@@ -0,0 +1,31 @@
+import zipTextNodes from 'zip-text-nodes';
+
+// Using https://www.conventionalcommits.org/ as a reference.
+
+export function getConventionalCommitAndScopeMatch(commitTitle: string): RegExpExecArray | undefined {
+	const match = /^(?<type>\w+)(?:\((?<scope>.+)\))?: .+$/.exec(commitTitle);
+
+	if (!match) {
+		return;
+	}
+
+	return match;
+}
+
+export function removeCommitAndScope(
+	element: HTMLElement,
+	match: RegExpExecArray,
+): void {
+	const {type, scope} = match?.groups ?? {};
+
+	const semanticCommitTypeAndScope = scope ? `${type}(${scope}):` : `${type}:`;
+
+	const modified = element.textContent.replace(semanticCommitTypeAndScope, match => `<s>${match}</s>`);
+
+	const temporaryDiv = document.createElement('div');
+	temporaryDiv.innerHTML = modified;
+
+	zipTextNodes(element, temporaryDiv);
+
```

Comment:
We can simplify this function, but it would drop support for `feat(#123): title`, which is totally fine for me. 

```js 
assertNodeContent(node, theRegexAbove)
node.textContent = node.textContent.replace(theRegexAbove)
```

Notes:

- you will likely need to find the exact text node for this to work safely, probably `node = element.firstChild`
- you can find the assertion function in our helpers 
- the regex from the other function should be moved to the top scope and not copy-pasted

Reply:
Sorry but i don't quite understand what you mean with this review.

Reply:
Those two lines are the entirety of the function, plus the likely `node = element.firstChild` line. Forget the zip dependency 🙌 👍

Reply:
Can you elaborate on what you mean by # in the scope?

Reply:
This solution seems to break here:

![image](https://github.com/user-attachments/assets/90c800d4-6e72-48cf-a718-1b044aebeb8b)

(Likely an issue with `node = element.firstChild`: 
![image](https://github.com/user-attachments/assets/dfa44242-5408-49a4-9edb-78794d99aa53))

Reply:
Try creating that # example commit. # is used for issue references, so GitHub will create this DOM:

``` html
<a href=“commit”>
  feat(
</a>
<a href=“issue or PR”>
  #123 
</a>
<a href=“commit”>
  ): title
</a>
``` 

Which is perfectly supported via zip-text-nodes but not with your first solution or this last solution. 

This is likely not a common scenario so that’s why I’m ok with it not being supported.